tar/bsdtar.c

```diff
@@ -1075,7 +1080,7 @@ main(int argc, char **argv)
 
 	/* Make sure we have whatever keys we're going to need. */
 	if (bsdtar->have_keys == 0) {
-		if (!bsdtar->option_dryrun) {
+		if ((!bsdtar->option_dryrun) && (!bsdtar->option_dump_config)) {
```

Comment:
Doesn't this mean that `tarsnap --dump-config` will generate (fake) keys?  I'm not sure I understand what you're doing here.